README.md

```diff
@@ -18,32 +18,57 @@ In order to run the tests in this repository, please clone each of the following
 
 1. Open a terminal and navigating to `[working folder]`
 2. Execute `docker network create -d bridge my-network`
 
-### Docker Compose
+### 5 Docker Compose
 
 To run the containers, you will need a Docker Compose file.
 
 1. Open a terminal and navigate cd to `[working folder]/Tests`
 3. Execute `docker compose up --build` to build and run the containers specified in the `docker-compose.yml` file. This has completed when you see "created" next to each of the 4 containers and you see logs in the terminal
 
+### 6 Run Test
+
+You can run the test in a terminal or Visual Studio. These steps are for a terminal but you can get a more detailed view by running the tests in test Explorer in Visual Studio.
+
+> The test will use the version of the code which is on your local device for each repository/container. Generally, you'll want to make sure you have pulled the latest changes on the `main` branch before running the test, although there may be scenarios where you want to test from other branches.
+
+1. Open a terminal and navigate to `[working folder]/Tests/src/`
+2. Execute `dotnet test`
+3. If successful, you will see that 2 tests from "Tests.IntegrationTests.dll" has "Passed!" in the terminal.
+
+## Running tests locally against Azure hosted environment
+
+You may want to run this test against deployed component in Azure. This can be usefull to make sure that the deployed components have the correct settings and can communicate correctly.
+
+In order to do this follow steps 1 and 2 from Running tests locally with Docker Compose and then follow these steps:
+
+### Configure secrets.json
+
+In order to target a deployed set of resources, the settings in `appsettings.json` need to be over-written with values that corespond to the deployed components. It is not good practice to edit `appsettings.json` because that file is under source control and you may inadvertently commit secrets to Git. Instead you can use .net user secrets to create a local secrets.json which over-rides the settings in `appsettings.json`. See https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-6.0
+
+1. Open a terminal and navigate to `[working folder]/Tests/src/`
+2. Execute `dotnet user-secrets init`
+3. Manually add equivilent secrets from the Azure-hosted components for each key/value pair in `[working folder]/Tests/src/Tests.IntegrationTests/appsettings.json` using this command `dotnet user-secrets set "setting key" "setting value"`
+   1. As an alternative, you can open `secrets.json` for this project in a text editor and add the secrets that way. By default, `secrets.json` will be located in `%APPDATA%\Microsoft\UserSecrets\<user_secrets_id>\secrets.json` for Windows or `~/.microsoft/usersecrets/<user_secrets_id>/secrets.json` for Linux/Mac. You can find the value of `<user_secrets_id>` by opening the `[working folder]/Tests/src/Tests.IntegrationTests/Tests.IntegrationTests.csproj` in a text editor.
+
```

Comment:
Is it worth a table to explain the settings in appsettings?  Or perhaps mention ```UseFixture``` and its usage in here?

Reply:
Have added a section to the doc to explain each settings and where to get the values from